### PR TITLE
Make sure we install rasa in non-editable mode in Dockerfiles

### PR DESCRIPTION
**Pull request by alwx**
**Proposed changes**:
- fix DistributionNotFound issue:
```
docker run -v $(pwd):/app rasa/rasa init --no-prompt
...
Status: Downloaded newer image for rasa/rasa:latest
Traceback (most recent call last):
  File "/opt/venv/bin/rasa", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/opt/venv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3253, in <module>
    @_call_aside
  File "/opt/venv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3237, in _call_aside
    f(*args, **kwargs)
  File "/opt/venv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3266, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/opt/venv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/opt/venv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/opt/venv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'rasa' distribution was not found and is required by the application
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -31,21 +31,23 @@ RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-
...
+WORKDIR /build
 
 # install dependencies
 RUN python -m venv /opt/venv && \
   . /opt/venv/bin/activate && \
   pip install --no-cache-dir -U 'pip<20' && \
-  cd /app && \
-  poetry install --no-dev --no-interaction
+  poetry install --no-dev --no-root --no-interaction && \
+  poetry run pip install .
```
Comment (degiz) [inline Dockerfile]: This step wasn't here before? Do we need it?

Comment (degiz) [reply Dockerfile]: I mean `poetry run pip install`

Comment (alwx) [reply Dockerfile]: Yes!
The idea is poetry by default installs your app in editable mode (when you just run `poetry install`). By doing `poetry install --no-root` we install everything except `rasa` itself, and `poetry run pip install .` installs rasa in non-editable mode using the already installed dependencies.

Comment (alwx) [reply Dockerfile]: There are other ways of doing the same (for example, https://stackoverflow.com/questions/53835198/integrating-python-poetry-with-docker) but they all do more or less the same in the end.

Comment (wochinge) [reply Dockerfile]: I think it should be `poetry pip install --no-deps` (https://github.com/python-poetry/poetry/issues/1382#issuecomment-588544067)

Comment (alwx) [reply Dockerfile]: One minute, will check it out

Comment (degiz) [reply Dockerfile]: looks like that.. @alwx do you know if pip would use the poetry lockfile?

Comment (degiz) [reply Dockerfile]: I'd guess that no

**Pull request merged**